apps/ledger-live-desktop/src/renderer/components/SyncOnboarding/Manual/index.tsx

```diff
@@ -92,26 +93,53 @@ const SyncOnboardingScreen: React.FC<SyncOnboardingScreenProps> = ({
   const { onboardingState, allowedError, fatalError, lockedDevice } = useOnboardingStatePolling({
     device: lastSeenDevice,
     pollingPeriodMs: POLLING_PERIOD_MS,
-    stopPolling: !isPollingOn || isBootloader === null || isBootloader,
+    stopPolling: !isPollingOn || isBootloader,
   });
 
   const { state: toggleOnboardingEarlyCheckState } = useToggleOnboardingEarlyCheck({
     deviceId: lastSeenDevice?.deviceId ?? "",
     toggleType: toggleOnboardingEarlyCheckType,
   });
 
-  const refreshIsBootloaderMode = useCallback(() => {
-    if (!device) return;
-    withDevice(device.deviceId)(transport => from(getDeviceInfo(transport)))
-      .toPromise()
-      .then((deviceInfo: DeviceInfo) => {
-        setIsBootloader(deviceInfo?.isBootloader);
-      });
-  }, [device]);
-
   useEffect(() => {
+    let dead = false;
+    function refreshIsBootloaderMode() {
```

Comment:
So even with the polling emitting when the device is in bootloader you need this function/to do a `getDeviceInfo` ? (genuinely asking, idk ahah)

Reply:
For instance if the fw update gets interrupted because the device gets unplugged, if the device is in bootloader mode, then once the user plugs it again, `device` changes and triggers this `useEffect`. And at that moment the polling is not running (because it's right after the fw update flow was interrupted) so it allows us to handle that situation and get the device out of bootloader again.